src/vivarium/component.py

```diff
@@ -92,7 +92,24 @@ class Component(ABC):
     component. An empty dictionary indicates no managed configurations.
     """
 
-    def __repr__(self):
+    def __init__(self) -> None:
```

Comment:
This is a directly copy/paste that moves the constructor to the top. mypy really didn't like the init being at the bottom of the file.